src/m365/spo/commands/site/site-chrome-set.spec.ts

```diff
@@ -0,0 +1,385 @@
+    });
+
+    command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/test', disableMegaMenu: "true" } }, () => {
+      try {
+        assert.strictEqual(data.megaMenuEnabled, false);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('Validate if footer gets disabled', (done) => {
+    let data: any = {};
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/test/_api/web/SetChromeOptions`) {
+        data = opts.data;
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/test', disableFooter: "true" } }, () => {
+      try {
+        assert.strictEqual(data.footerEnabled, false);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('Validate if title in header gets disabled', (done) => {
+    let data: any = {};
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/test/_api/web/SetChromeOptions`) {
+        data = opts.data;
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/test', hideTitleInHeader: "true" } }, () => {
+      try {
+        assert.strictEqual(data.hideTitleInHeader, true);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('Validate with custom values', (done) => {
+    let data: any = {};
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/test/_api/web/SetChromeOptions`) {
+        data = opts.data;
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/test', headerLayout: "Extended", headerEmphasis: "Light", logoAlignment: "Center", footerLayout: "Extended", footerEmphasis: "Light" } }, () => {
+      try {
+        assert.strictEqual(data.headerLayout, 4);
+        assert.strictEqual(data.headerEmphasis, 1);
+        assert.strictEqual(data.logoAlignment, 1);
+        assert.strictEqual(data.footerLayout, 2);
+        assert.strictEqual(data.footerEmphasis, 2);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+
+
+  it('correctly handles OData error when setting site chrome settings', (done) => {
+    sinon.stub(request, 'post').callsFake(() => {
+      return Promise.reject({ error: { 'odata.error': { message: { value: 'An error has occurred' } } } });
+    });
+
+    command.action(logger, { options: { debug: false, url: 'https://contoso.sharepoint.com/sites/Sales', sharingCapability: 'Invalid' } } as any, (err?: any) => {
```

Comment:
Theoretically, this case is impossible, because the `Invalid` value wouldn't pass validation which executes before the command code but I get your point :) Also, `sharingCapability` is not a valid option for this command

Reply:
Oops, must be a thing I forgot from the file copy :-)